File: app/src/main/java/com/example/somachar/News.java
package com.example.somachar;

import com.google.gson.annotations.SerializedName;

public class News {

    // Deserialize JSON
    @SerializedName("heading")
    private final String heading;
    @SerializedName("imagelink")
    private final String imagelink;
    @SerializedName("newslink")
    private final String newslink;
    @SerializedName("papername")
    private final String papername;
    @SerializedName("time")
    private final String time;
    @SerializedName("details")
    private final String details;

    // News class constructor
    public News(String heading, String imagelink, String newslink, String papername, String time, String details) {
        this.heading = heading;
        this.imagelink = imagelink;
        this.newslink = newslink;
        this.papername = papername;
        this.time = time;
        this.details = details;
    }

    public String getHeading() {
        return heading;
    }

    public String getImagelink() {
        return imagelink;
    }

    public String getNewslink() {
        return newslink;
    }

    public String getPapername() {
        return papername;
    }

    public String getTime() {
        return time;
    }

    public String getDetails() {
        return details;
    }
}

File: app/src/main/java/com/example/somachar/News.kt
package com.example.somachar


class News(
    private val heading: String,
    private val imageLink: String,
    private val newsLink: String,
    private val paperName: String,
    private val time: String,
    private val details: String
){

    fun getHeading(): String{
        return heading
    }
    fun getImageLink(): String{
        return imageLink
    }
    fun getNewsLink(): String{
        return newsLink
    }
    fun getPaperName(): String{
        return paperName
    }
    fun getTime(): String{
        return time
    }
    fun getDetails(): String{
        return details
    }
}
